### ISSUE TITLE
The search media search is not recursive

### ISSUE THREAD

**Issue by mamazu** [sulu/sulu#7152]
# Problem description
Let's assume we have the following structure in the media
```
media
    +-> webspaceKey1
    |----> test
        +-> a.png
    |----> other
        +> b.png
    +-> webspaceKey2
    |----> test
        +-> a.png
    |----> other
        +> b.png
```

If we just open the media site and search globally then we get two results for `a.png`. If we want to search only inside the folder `webspaceKey1` then you will find no results. If you then go into the right folder (aka `webspaceKey1/test` and search for it you will find it.

The problem is that the search once I click on a folder only filters the folders direct children.

# Proposed solution
It would be nice if you could recursively search in folders. This will however have an in impact on performance because we first have to calculate the tree of folders that the database has to search.

**Comment by alexander-schranz** [comment]
Currently we only return inside a collection media which are part of that collection. Showing also the child medias could be confusing without somebody actively doing an action. The Root currently shows all medias for quick access of the last media unrelated to collection as last uploaded can be accessed without navigating to the media collections.

So I think the only way to target that issue would be that we add additonal `Include ...` flag, beside the Search / Filters. Maybe we need to discuss this with the UI team how this should look like or if it even should be part of the Filters and how we handle the root collection then as there it does not make sense to have that filter in case of medias. Also we need to keep in mind how it behaves in other views.

**Comment by mamazu** [comment]
Yes, that would be a good idea to add a flag as the recursive search can be quite resource intense. If you have a design I'd be happy to provide the search api implementation.
